packages/sync/src/modules/Full_Sync.php

```diff
@@ -174,78 +174,145 @@ public function start( $module_configs = null ) {
 	 * @access public
 	 *
 	 * @param array $configs Full sync configuration for all sync modules.
-	 * @param array $enqueue_status Current status of the queue, indexed by sync modules.
 	 */
-	public function continue_enqueuing( $configs = null, $enqueue_status = null ) {
-		if ( ! $this->is_started() || $this->get_status_option( 'queue_finished' ) ) {
+	public function continue_enqueuing( $configs = null ) {
+		if ( ! $this->is_started()
+			 ||
+			 ! $this->attempt_enqueue_lock()
+			 ||
+			 $this->get_status_option( 'queue_finished' ) ) {
+
 			return;
 		}
 
-		// If full sync queue is full, don't enqueue more items.
-		$max_queue_size_full_sync = Settings::get_setting( 'max_queue_size_full_sync' );
-		$full_sync_queue          = new Queue( 'full_sync' );
-
-		$available_queue_slots = $max_queue_size_full_sync - $full_sync_queue->size();
+		$send_immediately = Settings::get_setting( 'full_sync_send_immediately' );
 
-		if ( $available_queue_slots <= 0 ) {
-			return;
+		if ( $send_immediately ) {
```

Comment:
You could join this with the line above and just do:

```php
if ( Settings::get_setting( 'full_sync_send_immediately' ) ) {
// ...
```

Reply:
https://github.com/Automattic/jetpack/pull/13963/commits/88448dc0a11229e12ac60237e8bd01603c6a701f